spec.html

```diff
@@ -36112,8 +36385,24 @@ <h1>
         </emu-alg>
         <emu-grammar>AtomEscape :: CharacterClassEscape</emu-grammar>
         <emu-alg>
-          1. Let _A_ be CompileToCharSet of |CharacterClassEscape| with argument _rer_.
-          1. Return CharacterSetMatcher(_rer_, _A_, *false*, _direction_).
+          1. Let _cs_ be CompileToCharSet of |CharacterClassEscape| with argument _rer_.
+          1. If _rer_.[[UnicodeSets]] is *false*, or if every element of _cs_ consists of a single character (including if _cs_ is empty), return CharacterSetMatcher(_rer_, _cs_, *false*, _direction_).
+          1. Let _lm_ be an empty List of Matchers.
+          1. For each string _s_ in _cs_ whose length is greater than 1, iterating in descending order of string length, do
```

Comment:
Should this be manipulating CharSetElements instead of strings and doing manual code point -> code unit manipulation? I haven't done a thorough review so it might actually be okay, though it seems questionable to me to allow this kind of interchangeability between CharSetElements and String values.

Reply:
Hmm, this hasn’t come up until now, and I’m inclined to think this is okay as-is, but I defer to you and the other editors.

We iterate over a `CharSet`’s members in similar ways elsewhere. For example, see argument `A` (a `CharSet`) in `CharacterSetMatcher`, and how it’s used in step `l`:

```
l. If there exists a member _a_ of _A_ such that Canonicalize(_rer_, _a_) is _cc_, let _found_ be *true*. Otherwise, let _found_ be *false*.
```

We could change that to explicitly manipulate `CharSetElement`:

```
l. If there exists a CharSetElement _cse_ of _A_ containing exactly one character _a_ such that Canonicalize(_rer_, _a_) is _cc_, let _found_ be *true*. Otherwise, let _found_ be *false*.
```

Would that be an improvement?

Reply:
Your concrete alternative reads better to me and would be preferred. Now that I understand it better it's fine to use strings interchangeably but I'd also appreciate a sentence somewhere about this being a thing that the spec does.

Reply:
PTAL: https://github.com/mathiasbynens/ecma262/pull/28/files

Reply:
I’ll go ahead and merge this, but feel free to leave review comments after the fact — I’ll still address them :)